pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1583,22 +1381,13 @@ func (og *operationGenerator) GenerateExpandInUseVolumeFunc(
 	volumeToMount VolumeToMount,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater) (volumetypes.GeneratedOperations, error) {
 
-	fsResizeFunc := func() (error, error) {
-		// Need to translate the spec here if the plugin is migrated so that the metrics
-		// emitted show the correct (migrated) plugin
-		if useCSIPlugin(og.translator, og.volumePluginMgr, volumeToMount.VolumeSpec) {
-			csiSpec, err := translateSpec(og.translator, volumeToMount.VolumeSpec)
-			if err != nil {
-				return volumeToMount.GenerateError("NodeExpandVolume.translateSpec failed", err)
-			}
-			volumeToMount.VolumeSpec = csiSpec
-		}
-		volumePlugin, err :=
-			og.volumePluginMgr.FindPluginBySpec(volumeToMount.VolumeSpec)
-		if err != nil || volumePlugin == nil {
-			return volumeToMount.GenerateError("NodeExpandVolume.FindPluginBySpec failed", err)
-		}
+	volumePlugin, err :=
+		og.volumePluginMgr.FindPluginBySpec(volumeToMount.VolumeSpec)
```

Comment:
Just confirming: is it OK that we are calling this earlier? It used to be invoked when `fsResizeFunc` is called. It will cause the same error to come from a different function.

Reply:
Yes this change is intentional. The block here essentially maps to the code [below](https://github.com/kubernetes/kubernetes/pull/83098/files/259f931651ae99711db2b95c1a56a9d55d923e1a#diff-450e811a4953f760ff1594ede8b2037eL1671) which did return the error we return above.
The extra `FindPluginBySpec` at the point of invocation of the generated fsResizeFunc is not necessary as we don't need `useCSIPlugin` at the point of invocation.